docs/guides/quickfort-alias-guide.rst

```diff
@@ -0,0 +1,850 @@
+.. _quickfort-alias-guide:
+
+Quickfort Alias Guide
+=====================
+
+Aliases allow you to use simple words to represent complicated key sequences
+when configuring buildings and stockpiles in quickfort ``#query`` blueprints.
+
+For example, say you have the following ``#build`` and ``#place`` blueprints:
+
+::
+
+    #build masonry workshop
+    ~, ~,~,`,`,`
+    ~,wm,~,`,`,`
+    ~, ~,~,`,`,`
+
+    #place stockpile for mason
+    ~,~,~,s,s,s
+    ~,~,~,s,s,s
+    ~,~,~,s,s,s
+
+and you want to configure the stockpile to hold only non-economic ("other")
+stone and to give to the adjacent mason workshop. You could write the
+key sequences directly:
+
+::
+
+    #query configure stockpile with expanded key sequences
+    ~,~,~,s{Down 5}deb{Right}{Down 2}p^,`,`
+    ~,~,~,g{Left 2}&,                   `,`
+    ~,~,~,`,                            `,`
+
+or you could use aliases:
+
+::
+
+    #query configure stockpile with aliases
+    ~,~,~,otherstone,`,`
+    ~,~,~,give2left, `,`
+    ~,~,~,`,         `,`
+
+If the stockpile had only a single tile, you could also replay both aliases in
+a single cell:
+
+::
+
+    #query configure mason with multiple aliases in one cell
+    ~,~,~,{otherstone}{give2left},`,`
+    ~,~,~,`,                      `,`
+    ~,~,~,`,                      `,`
+
+With aliases, blueprints are much easier to read and understand. They also
+save you from having to copy the same long key sequences everywhere.
+
+Alias definition files
+----------------------
+
+DFHack comes with a library of aliases for you to use that are always
+available when you run a ``#query`` blueprint. Many blueprints can be built
+with just those aliases. This "standard alias library" is stored in
+:source:`data/quickfort/aliases-common.txt`. The aliases in that file are
+described at the `bottom of this document <quickfort-alias-library>`.
+
+Please do not edit the aliases in the standard library directly. The file will
+get overwritten when DFHack is updated and you'll lose your changes. Instead,
+add your custom aliases to :source:`dfhack-config/quickfort/aliases.txt`.
+Definitions in this file take precedence over any definitions in the standard
+library.
+
+Alias syntax and usage
+----------------------
+
+The syntax for defining aliases is:
+
+::
```

Comment:
There is a shorthand you can use in cases like these (not mandatory, of course) - ending the line above with a double colon will render a single colon _and_ begin a code block. https://raw.githubusercontent.com/DFHack/dfhack/develop/docs/Compile.rst has some examples of this.

There's also a more explicit form, `.. code-block::`, that you can use to set syntax highlighting on a per-block basis, but the blocks in this document all seem to be displaying reasonably with the default "DFHack" highlighting (which basically just styles lines starting with `#` differently).

Reply:
thank you for this tip! several hundred superfluous lines are now deleted.

for the ``.. code-block::`` format, how do you set the language for the syntax highlighting? I believe the default is python, which isn't always appropriate

Reply:
`.. code-block:: language`
I made a couple changes in 09f262a0c234ad34da176ffb6bd9cd125384572a and a5f85e256d5f443bbc979969b37bc268122fb6e3
I think the default looks good here, but you can also change the default for all following code blocks with the `.. highlight:: language` directive (Compile.rst sets this to `shell` initially)